collectfast/strategies/boto3.py

```diff
@@ -29,6 +34,28 @@ def _clean_hash(quoted_hash: Optional[str]) -> Optional[str]:
         assert quoted_hash[0] == quoted_hash[-1] == '"'
         return quoted_hash[1:-1]
 
+    def get_local_multipart_hash(
+        self, path: str, local_storage: Storage, chunk_size: int
+    ) -> str:
+        """Calculate multipart hash using a given chunk size."""
+        chunk_hashes = []
+        with local_storage.open(path, "rb") as f:
+            func: Callable[[], Any] = lambda: f.read(chunk_size)
```

Comment:
I'd prefer if we can define this function using `def` instead of `lambda`, and also with it's actual signature instead `Any`.

Reply:
I move it to a `def` instead of `lambda` in 41c89cc, but I'm pretty new to the type hints and after a bit of trying I can't figure out what it should be if not `Any`. I keep getting weird errors when I try what I think the return value should be.

Reply:
N/m figured out a better implementation that doesn't require a function.